play.pokemonshowdown.com/src/battle-team-editor.tsx

```diff
@@ -1350,6 +1353,14 @@ class TeamWizard extends preact.Component<{
 			switch (type) {
 			case 'pokemon':
 				set.species = name;
+				const species = editor.dex.species.get(name);
+				if (!(editor.format.includes('hackmons') || editor.format.endsWith('bh')) && species.requiredItems.length === 1) {
+					set.item = species.requiredItems[0];
+				} else {
+					set.item = '';
+				}
+				const abilities = Object.values(species.abilities);
+				if (abilities.length === 1) set.ability = abilities[0];
```

Comment:
```suggestion
				set.ability = abilities.length === 1 ? abilities[0] : '';
```

It's probably worth clearing the ability if not.

Reply:
```suggestion
				if (abilities.length === 1) set.ability = abilities[0];
				else if (!Object.values(abilities).includes(set.ability)) set.ability = '';
```

There, a lil smarter.

Reply:
You know, probably we should check for hackmons or AAA on the ability check, too...